src/input/mod.rs

```diff
@@ -2344,17 +2380,29 @@ fn should_intercept_key(
                     repeat: true,
                     cooldown: None,
                     allow_when_locked: false,
+                    // The screenshot UI owns the focus anyays, so this doesn't really matter?
+                    // But logically, nothing can inhibit its actions. Only opening it can be
+                    // inhibited.
+                    allow_inhibiting: false,
                 });
             }
         }
     }
 
     match (final_bind, pressed) {
         (Some(bind), true) => {
-            suppressed_keys.insert(key_code);
-            FilterResult::Intercept(Some(bind))
+            if is_inhibiting_shortcuts && bind.allow_inhibiting {
+                FilterResult::Forward
+            } else {
+                suppressed_keys.insert(key_code);
+                FilterResult::Intercept(Some(bind))
+            }
         }
         (_, false) => {
+            // By this point, we know that the key was supressed on press. Even if we're inhibiting
+            // shortcuts, we should still suppress the release. But we don't need to check for
+            // shortcuts inhibition here, because if it wasn't inhibited on press, it wouldn't be
+            // suppressed, so it would already have been forwarded at the start of this function.
```

Comment:
```suggestion
            // shortcuts inhibition here, because if it was inhibited on press (forwarded to the client), it wouldn't be
            // suppressed, so it would already have been forwarded at the start of this function.
```
Is this right?

Reply:
yeah. that seems correct. i think i wrote that thinking "if it wasn't intercepted" (== YES inhibited). or thinking that the keypress itself is inhhibited; but what that really means here is that the compositor's handling of the keypress is inhibited.